esphome/components/light/color_mode.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <cstdint>
+
+namespace esphome {
+namespace light {
+
+/// Color channels are the various outputs that a light has and that can be independently controlled by the user.
+enum class ColorChannel : uint8_t {
+  /// Intensity of a single white output.
+  WHITE = 1,
```

Comment:
```suggestion
  MAIN = 1,
```

Less clear than in `ColorMode`, but I think maybe naming it that way would be better?